src/RPC/MultiRPC.php

```diff
@@ -0,0 +1,404 @@
+
+    /**
+     * @param array<int, RelayInterface> $relays
+     */
+    public function __construct(
+        array $relays,
+        CodecInterface $codec = new JsonCodec()
+    ) {
+        foreach ($relays as $relay) {
+            if (!($relay instanceof SocketRelay)) {
+                throw new RPCException("MultiRPC can only be used with sockets, no pipes allowed");
+            }
+        }
+
+        $this->freeRelays = $relays;
+        parent::__construct($codec);
+    }
+
+    /**
+     * Without cloning relays explicitly they get shared and thus, when one RPC gets called, the freeRelays array
+     * in the other RPC stays the same, making it reuse the just-used and still occupied relay.
+     */
+    public function __clone()
+    {
+        foreach ($this->freeRelays as $key => $relay) {
+            $this->freeRelays[$key] = clone $relay;
+        }
+        foreach ($this->occupiedRelays as $key => $relay) {
+            $this->occupiedRelays[$key] = clone $relay;
+        }
+        $this->seqToRelayMap = $this->asyncResponseBuffer = [];
+    }
+
+    /**
+     * @param non-empty-string $connection
+     * @param positive-int $count
+     */
+    public static function create(string $connection, int $count = 50, CodecInterface $codec = new JsonCodec()): self
+    {
+        assert($count > 0);
+        $relays = [];
+
+        for ($i = 0; $i < $count; $i++) {
+            $relays[] = Relay::create($connection);
+        }
+
+        return new self($relays, $codec);
+    }
+
+    /**
+     * Force-connects all SocketRelays.
+     * Does nothing if no SocketRelay.
+     */
+    public function preConnectRelays(): void
+    {
+        if (count($this->freeRelays) === 0) {
+            return;
+        }
+
+        if (!$this->freeRelays[0] instanceof SocketRelay) {
+            return;
+        }
+
+        foreach ($this->freeRelays as $relay) {
+            assert($relay instanceof SocketRelay);
+            // Force connect
+            $relay->connect();
+        }
+    }
+
+
+    public function call(string $method, mixed $payload, mixed $options = null): mixed
+    {
+        $relayIndex = $this->ensureFreeRelayAvailable();
+        $relay = $this->freeRelays[$relayIndex];
+
+        $relay->send($this->packFrame($method, $payload));
+
+        // wait for the frame confirmation
+        $frame = $this->getResponseFromRelay($relay, self::$seq, true);
+
+        self::$seq++;
+
+        return $this->decodeResponse($frame, $relay, $options);
+    }
+
+    public function callIgnoreResponse(string $method, mixed $payload): void
+    {
+        $relayIndex = $this->ensureFreeRelayAvailable();
+        $relay = $this->freeRelays[$relayIndex];
+
+        $relay->send($this->packFrame($method, $payload));
+
+        $seq = self::$seq;
+        self::$seq++;
+        $this->occupiedRelays[$seq] = $relay;
+        // Last index so no need for array_pop or stuff
+        unset($this->freeRelays[$relayIndex]);
```

Comment:
I think it complains about the declared `@throws \Error` in SocketRelay::connect().
Afaict that Error should never be thrown since `extension_loaded(sockets)` is checked in the constructor. Lmk if I can remove that

Reply:
@msmakouz Following this exchange and reading the docs more I've removed the annotation in commit c4b41cff